websites/D/Desu-Online/presence.ts

```diff
@@ -25,135 +35,155 @@ presence.on("UpdateData", async () => {
-		presenceData.details = "Przegląda politykę prywatności...";
-	else if (pathname.startsWith("/genres/")) {
-		presenceData.details = "Przegląda gatunek:";
-		presenceData.state = `${
-			document.querySelector("div.releases > h1 > span").textContent
-		}`;
-		if (pathname.includes("page"))
-			presenceData.state += ` (Strona ${pathname.split("/")[4]})`;
-	} else if (pathname.startsWith("/studio/")) {
-		presenceData.details = "Przegląda studio:";
-		presenceData.state = `${
-			document.querySelector("div.releases > h1 > span").textContent
-		}`;
-		if (pathname.includes("page"))
-			presenceData.state += ` (Strona ${pathname.split("/")[4]})`;
-	} else if (
-		document.querySelector("span.year").textContent &&
-		document.querySelector("span.year").textContent.includes("Dodane przez")
-	) {
-		presenceData.details = `Ogląda: ${
-			document.querySelector("div.det > h3").textContent
-		}`;
-		const playinfo = document
-			.querySelector("li.selected > a > div.playinfo > span")
-			.textContent.split("-");
-		presenceData.state = `Odcinek:${playinfo[0].replace("Odc", "")}`;
-		if (playinfo.length > 2) {
-			presenceData.state += ` - "${document
-				.querySelector("li.selected > a > div.playinfo > span")
-				.textContent.match(/-.*-/gm)
-				.toString()
-				.slice(2, -2)}"`;
-		}
-		presenceData.buttons = [
-			{ label: "Oglądaj", url: document.URL },
-			{
-				label: "Cała seria",
-				url: document.querySelector("div.det > h3 > a").getAttribute("href"),
-			},
-		];
-		presenceData.largeImageKey = document
-			.querySelector("img.ts-post-image")
-			.getAttribute("src");
-		[presenceData.startTimestamp, presenceData.endTimestamp] =
-			presence.getTimestamps(video.current, video.duration);
+	} else {
+		switch (pathname) {
+			case pathname.startsWith("/anime/") ? pathname : null:
+				if (pathname === "/anime/") presenceData.details = "Przegląda serie...";
+				else if (pathname === "/anime/list-mode/")
+					presenceData.details = "Przegląda listę serii...";
+				else {
+					presenceData.details = "Przegląda serię";
+					presenceData.state =
+						document.querySelector("h1.entry-title").textContent;
+					presenceData.largeImageKey = document
+						.querySelector("img.ts-post-image")
+						.getAttribute("src");
+					presenceData.buttons = [
+						{
+							label: "Odwiedź",
+							url: document.URL,
+						},
+					];
+				}
+				break;
+			case pathname.startsWith("/blog/") ? pathname : null:
+				if (pathname === "/blog/") {
+					presenceData.details = "Przegląda bloga";
+					presenceData.state = "Strona 1";
+				} else if (pathname.startsWith("/blog/page/")) {
+					presenceData.details = "Przegląda bloga:";
+					presenceData.state = `Strona ${pathname.split("/")[3]}`;
+				} else {
+					presenceData.details = "Czyta bloga";
+					presenceData.state = document
+						.querySelectorAll("h1")[0]
+						.textContent.replace("\n", "");
+					presenceData.buttons = [{ label: "Czytaj", url: document.URL }];
```

Comment:
Use `document.location.href` instead of `document.URL` for consistency
```suggestion
					presenceData.buttons = [{ label: "Czytaj", url: href }];
```